### ISSUE TITLE
Bug com as datas

### ISSUE THREAD

**Issue by fernao** [RedeMocambos/baobaxia#44]
Publiquei um arquivo à 0:20 do dia 22; ele entrou como do dia anterior (21). Talvez fuso, não sei

**Comment by agger-magenta** [comment]
Is this still a problem? It could be due  to different time settings on client and server. In that case, it probably can't be fixed in the code.

**Comment by agger-magenta** [comment]
I can confirm this bug.
It's now 00:17, and I published a media at hyndla and it was published as if today was the 20th - it should be 21.

**Comment by agger-magenta** [comment]
The timestamp shown in the GUI is this:

2015-04-20T22:14:24.669Z

The time on this computer is correct, however, and the time on the server is also correct - and we're in the same time zone.  Maybe all times are set to be two hours early. If so, this should not be too hard to locate.

**Comment by fernao** [comment]
yo carsten,

how are you?

we're at tainã right now, if you can, join #mocambos at freenode, I'll 
be there!

fernão

Em 2015-04-20 19:21, Carsten Agger escreveu:

**Comment by agger-magenta** [comment]
Done, but only briefly, it's late :)

**Comment by agger-magenta** [comment]
I think I have a fix for this, but I'll have to wait til after midnight to test it. 

Note, the media creation date shown in the GUI is UTC. It should be local time. Is that a JavaScript or a serialization thing? It's shown correctly as local time in the Django admin.

**Comment by agger-magenta** [comment]
This fix in media/models.py will solve the problem, or rather: is required to solve the problem:

```
 def get_file_path(instance):
     """Retorna o caminho do media"""
     return os.path.join(REPOSITORY_DIR, get_media_path(instance))
@@ -58,7 +68,7 @@ def get_file_path(instance):
 def get_media_path(instance):
     # FIX: se mudar a data quebra o path
     if instance.date == '':
-        t = datetime.now
+        t = get_now()
         date = t.strftime("%y/%m/%d/")
     else:        
         if isinstance(instance.date, unicode):

@@ -68,7 +78,7 @@ def get_media_path(instance):
             date = date.strftime("%y/%m/%d/")
             """        date = datetime(instance.date)"""
         else:
-            date = instance.date.strftime("%y/%m/%d/")
+            date = timezone.localtime(instance.date).strftime("%y/%m/%d/")
```

... get_now() returns a local datetime.

However.

This will make get_file_path() fail for all media currently afflicted by this bug.

_And_ mucuas with different timezones will disagree on some URLs. We might want to store the file path and only calculate it when creating (or moving) media.

**Comment by agger-magenta** [comment]
And in fact this caused process-requests to fail because of some media affected by the bug, i.e. with wrong file paths.

If we change this, we need to find out how to update the existing media. Move them to the correct date, maybe.

**Comment by agger-magenta** [comment]
Or maybe stop using dates for the file path.

It could also be like /data/bbx/<repository>/<mucua>/<type>/YYYY/<UUID>/<files>

which would make it a lot easier to change dates too. Or maybe some other scheme entirely.

**Comment by agger-magenta** [comment]
A non-date-based scheme like this might work better:

/data/bbx/repositories/<network>/<mucua>/<type>/<user>

Like I wrote previously, I don't think the date thing can be solved satisfactorily unless we can assume mucuas are always in the same time zone. But there are many time zones in Brasil.

The drawback is that this requires a rather heavy migration.

**Comment by fernao** [comment]
yes... specially because now we have more than 20 mucuas currently working.

What we're doing right now is to fix, changing manually the dates of medias with broken date. But this may fail in the case you told (several time zones at same repository/network).

Maybe we can assume that we'll save all content only using GMT 0 (fixed timezone), and set a timezone for each mucua at settings.py. So, during date exhibition, we'll parse the correct date.

**Comment by agger-magenta** [comment]
Saving all content using GMT 0 always would be the best possible solution, I think. And always use TZ-aware datetimes so the time & date can be displayed correctly to the user.

**Comment by agger-magenta** [comment]
It seems they are indeed stored as GMT 0 now - at least this one was: http://baobaxia.modspil.dk/#mocambos/hyndla/media/4435ca80-9519-49e1-8db1-1fdfd8c313e6

But does this solve the problem?

**Comment by 20** [comment]
Agora pega o UTC!

**Comment by 20** [comment]
Django object date is still incoherent sometime with path
